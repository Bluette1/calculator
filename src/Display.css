.display {
  height: 100px;
  background-color: rgb(128, 128, 128);
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding: 10px;
  color: rgb(255, 255, 255);
  font-weight: bold;
}

@media (max-width: 400px) {
  .display {
    display: flex;
    justify-content: space-evenly;
    height: 80px;
  }
}
