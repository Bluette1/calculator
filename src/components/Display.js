import React from 'react';
import PropTypes from 'prop-types';
import '../Display.css';

const Display = props => {
  const { result, error } = props;
  let display = result;
  if (error) {
    display = '\u26A0';
  }
  return <div data-testid="display" className="display">{display}</div>;
};
Display.defaultProps = {
  result: '0',
  error: false,
};

Display.propTypes = {
  result: PropTypes.string,
  error: PropTypes.bool,
};

export default Display;
