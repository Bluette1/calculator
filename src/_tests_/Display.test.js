import React from 'react';
import ReactDOM from 'react-dom';
import { cleanup } from '@testing-library/react';

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Display from '../components/Display';

configure({ adapter: new Adapter() });

afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Display />, div);
});

configure({ adapter: new Adapter() });
it('renders the app and displays correct values', () => {
  const display = mount(<Display />);
  display.setProps({ result: '12' });
  expect(display.find('.display').text()).toBe('12');
  expect(display).toMatchSnapshot();
});

it('renders the app and displays correct values', () => {
  const display = mount(<Display />);
  display.setProps({ error: true });
  expect(display.find('.display').text()).toBe('\u26A0');
  expect(display).toMatchSnapshot();
});
