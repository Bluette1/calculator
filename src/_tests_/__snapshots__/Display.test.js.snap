// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app and displays correct values 1`] = `ReactWrapper {}`;

exports[`renders the app and displays correct values 2`] = `ReactWrapper {}`;
